engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/NadelResultTransformer.kt

```diff
@@ -226,7 +285,32 @@ private fun JsonMap.getJsonMapAt(path: JsonNodePath): JsonMap? {
 }
 
 private fun JsonMap.getAt(path: JsonNodePath): JsonNode {
-    return path.segments.fold(JsonNode(JsonNodePath.root, this), JsonNode::get)
+    var cursor: Any = this
```

Comment:
Not feature flagging this, if it's broken tests would've caught onto it. It's a simple function.